examples/applications/plot_digits_denoising.py

```diff
@@ -0,0 +1,139 @@
+    X, y, stratify=y, random_state=0, train_size=1_000, test_size=100
+)
+
+rng = np.random.RandomState(0)
+noise = rng.normal(scale=0.25, size=X_test.shape)
+X_test_noisy = X_test + noise
+
+noise = rng.normal(scale=0.25, size=X_train.shape)
+X_train_noisy = X_train + noise
+
+# %%
+# In addition, we will create a helper function to qualitatively assess the
+# image reconstruction by plotting the test images.
+import matplotlib.pyplot as plt
+
+
+def plot_digits(X, title):
+    """Small helper function to plot 100 digits."""
+    fig, axs = plt.subplots(nrows=10, ncols=10, figsize=(8, 8))
+    for img, ax in zip(X, axs.ravel()):
+        ax.imshow(img.reshape((16, 16)), cmap="Greys")
+        ax.axis("off")
+    fig.suptitle(title, fontsize=24)
+
+
+# %%
+# In addition, we will use the mean squared error (MSE) to quantitatively
+# assess the image reconstruction.
+#
+# Let's first have a look to see the difference between noise-free and noisy
+# images. We will check the test set in this regard.
+plot_digits(X_test, "Uncorrupted test images")
+plot_digits(X_test_noisy,
+            f"Noisy test images\n"
+            f"MSE: {np.mean((X_test - X_test_noisy) ** 2):.2f}")
+
+# %%
+# Learn the `PCA` basis
+# ---------------------
+#
+# We can now learn our PCA basis using both a linear PCA and a kernel PCA that
+# uses a radial basis function (RBF) kernel.
+from sklearn.decomposition import PCA, KernelPCA
+
+pca = PCA(n_components=32)
+kernel_pca = KernelPCA(n_components=400, kernel="rbf", gamma=1e-3,
+                       fit_inverse_transform=True, alpha=5e-3)
+
+pca.fit(X_train_noisy)
+_ = kernel_pca.fit(X_train_noisy)
+
+# %%
+# Reconstruct and denoise test images
+# -----------------------------------
+#
+# Now, we can transform and reconstruct the noisy test set. Since we used less
+# components than the number of original features, we will get an approximation
+# of the original set. Indeed, by dropping the components explaining less
+# variance in PCA, we hope to remove noise. Similar thinking happen in kernel
+# PCA; however, we expect a better reconstruction because we use a non-linear
+# kernel to learn the PCA basis and a kernel ridge to learn the mapping
+# function.
+X_reconstructed_kernel_pca = kernel_pca.inverse_transform(
+    kernel_pca.transform(X_test_noisy))
+X_reconstructed_pca = pca.inverse_transform(pca.transform(X_test_noisy))
+
+# %%
+plot_digits(X_test, "Uncorrupted test images")
+plot_digits(X_reconstructed_pca,
+            f"PCA reconstruction\n"
+            f"MSE: {np.mean((X_test - X_reconstructed_pca) ** 2):.2f}")
+plot_digits(X_reconstructed_kernel_pca,
+            f"Kernel PCA reconstruction\n"
+            f"MSE: {np.mean((X_test - X_reconstructed_kernel_pca) ** 2):.2f}")
+
+# %%
+# PCA has a lower MSE than kernel PCA. However, the qualitative might not favor
+# PCA instead of kernel PCA. We observe that kernel PCA is able to remove
+# background noise and provide a smoother image.
```

Comment:
```suggestion
# PCA has a lower MSE than kernel PCA. However, the qualitative analysis might
# not favor PCA instead of kernel PCA. We observe that kernel PCA is able to 
# remove background noise and provide a smoother image.
```